lib/task_helpers/development/replication.rb

```diff
@@ -0,0 +1,129 @@
+
+          # TODO: We have the technology to watch for this and report when it's all good or bad
+          puts "Local replication is setup... try checking for users in all regions: psql -U root development_replication_99 -c \"select id from users;\""
+        ensure
+          restore
+        end
+
+        def teardown
+          REMOTES.each do |r|
+            teardown_global_subscription_for_region(r)
+            teardown_remote_publication(r)
+          end
+        end
+
+        def setup_remote(region)
+          create_region(region)
+          configure_remote_region(region)
+        end
+
+        def setup_global(region)
+          create_region(region)
+          configure_global_region(region)
+        end
+
+        def database(region)
+          "#{DB_PREFIX}_#{region}"
+        end
+
+        # Example: DATABASE_URL='postgres://root:smartvm@localhost:5432/development_replication_99'
+        def database_url(region)
+          "postgres://#{PG_USER}:#{PG_PASS}@#{PG_HOST}:#{PG_PORT}/#{database(region)}"
+        end
+
+        def guid_file
+          GUID_FILE
+        end
+
+        def backup_guid
+          BACKUP_GUID
+        end
+
+        def configure_global_region_script
+          subs = []
+          REMOTES.each do |r|
+            subs << PglogicalSubscription.new(:host => PG_HOST, :port => PG_PORT, :user => PG_USER, :dbname => database(r), :password => PG_PASS)
+          end
+          MiqPglogical.save_global_region(subs, [])
+        end
+
+        private
+
+        def command_environment(region)
+          {"REGION" => region.to_s, "RAILS_ENV" => "development", "DATABASE_URL" => database_url(region)}
+        end
+
+        def configure_global_region(region)
+          run_command("bin/rails r 'TaskHelpers::Development::Replication.configure_global_region_script'", env: command_environment(region))
+          run_command("psql -U #{PG_USER} #{database(region)} -c 'select * from pg_subscription;'")
+        end
+
+        def configure_remote_region(region)
+          run_command("bin/rails r 'MiqRegion.replication_type= :remote'", env: command_environment(region))
+          run_command("psql -U #{PG_USER} #{database(region)} -c 'select * from pg_publication;'")
+        end
+
+        def create_region(region)
+          run_command("DISABLE_DATABASE_ENVIRONMENT_CHECK='true' bin/rake evm:db:region", env: command_environment(region))
+          run_command("bin/rails r 'EvmDatabase.seed_primordial'", env: command_environment(region))
+        ensure
+          FileUtils.rm_f(guid_file)
+        end
+
+        def run_command(command, raise_on_error: true, env: {})
+          puts "+" * 50
+          puts "Running: #{command.inspect} with env: #{env.inspect}..."
+          puts AwesomeSpawn.run!(command, env: env).output
+        rescue AwesomeSpawn::CommandResultError => err
+          raise if raise_on_error
+
+          puts "Error skipped: #{err}"
+        ensure
+          puts "-" * 50
```

Comment:
My eyes 👀 

How about this:

```suggestion
        def run_command(command, raise_on_error: true, env: {})
          puts "+ #{command.inspect} with env: #{env.inspect}...".green
          success = system(env, command)
          puts
          raise if raise_on_error && !success
```

If `.green` is not available, you can also just do

```ruby
puts "\e[32m#{command.inspect} with env: #{env.inspect}...\e[0m"
```

Reply:
I guess the only other thing that might make it more readable would be to not even show the command at all but instead print what the command is doing...  For example:

Instead of

`Running: "DISABLE_DATABASE_ENVIRONMENT_CHECK='true' bin/rake evm:db:region" with env: {"REGION"=>"2", "RAILS_ENV"=>"development", "DATABASE_URL"=>"postgres://root:smartvm@localhost:5432/development_replication_2"}...
Resetting development database to region 2...`

Perhaps

`Creating database for region 2 (name: development_replication_2)...`

Reply:
Also, if you want to highlight errors, you can do something like:

```ruby
unless success
  if raise_on_error
    puts "An error occurred during execution".red
    raise
  else
    puts "An error occurred during execution...skipping"...yellow
  end
```

Reply:
Done, see description.  It was a great idea to use color!  🟢 🔴 🟡